src/main/java/gift/service/WishService.java

```diff
@@ -2,33 +2,45 @@
 
 import gift.dto.WishRequestDto;
 import gift.dto.WishResponseDto;
+import gift.entity.Member;
 import gift.entity.Product;
+import gift.entity.Wish;
 import gift.exception.DuplicateWishException;
 import gift.exception.ResourceNotFoundException;
+import gift.repository.MemberRepository;
 import gift.repository.ProductRepository;
 import gift.repository.WishRepository;
 import java.util.List;
 import java.util.stream.Collectors;
 import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
 
 @Service
 public class WishService {
     private final WishRepository wishRepository;
     private final ProductRepository productRepository;
+    private final MemberRepository memberRepository;
 
-    public WishService(WishRepository wishRepository, ProductRepository productRepository) {
+    public WishService(
+            WishRepository wishRepository,
+            ProductRepository productRepository,
+            MemberRepository memberRepository
+    ) {
         this.wishRepository = wishRepository;
         this.productRepository = productRepository;
+        this.memberRepository = memberRepository;
     }
 
-    public List<WishResponseDto> getWishes(Long memberId){
-        return wishRepository.findByMemberId(memberId).stream()
-                .map(w -> {
-                    Product product = productRepository.findProductById(w.getProductId())
-                            .orElseThrow(() -> new ResourceNotFoundException("상품 정보를 찾을 수 없습니다: ID " + w.getProductId()));
+    @Transactional(readOnly = true)
+    public List<WishResponseDto> getWishes(Long memberId) {
+        Member member = memberRepository.findById(memberId)
+                .orElseThrow(() -> new ResourceNotFoundException("해당 회원을 찾을 수 없습니다. ID: " + memberId));
 
+        return wishRepository.findByMember(member).stream()
```

Comment:
이 방식이 좋다는 것은 아니지만 참고삼아 남기는 코멘트이니 가볍게 참고만해주세요. 😉

```suggestion
        return wishRepository.findByMemberId(memberId).stream()
```

위와 같이 JPQL을 활용해서 wish를 memberId로 바로 찾을 수도 있을 것 같아요. 

다만 지금과 같은 방식은 member가 없는건지 위시가 없는건지가 명확하지 않아서 상황에따라서는 적절하지 않을 수 있다고 생각해요. 😃